docs/resources/server/tags.md

```diff
@@ -1,83 +1,240 @@
+|----------------------------|----------------------------------------|
+| `BannerPattern`            | `BannerPatternTagsProvider`            |
+| `Biome`                    | `BiomeTagsProvider`                    |
+| `Block`                    | `BlockTagsProvider`                    |
+| `CatVariant`               | `CatVariantTagsProvider`               |
+| `DamageType`               | `DamageTypeTagsProvider`               |
+| `EntityType`               | `EntityTypeTagsProvider`               |
+| `FlatLevelGeneratorPreset` | `FlatLevelGeneratorPresetTagsProvider` |
+| `Fluid`                    | `FluidTagsProvider`                    |
+| `GameEvent`                | `GameEventTagsProvider`                |
+| `Instrument`               | `InstrumentTagsProvider`               |
+| `Item`                     | `ItemTagsProvider`                     |
+| `PaintingVariant`          | `PaintingVariantTagsProvider`          |
+| `PoiType`                  | `PoiTypeTagsProvider`                  |
+| `Structure`                | `StructureTagsProvider`                |
+| `WorldPreset`              | `WorldPresetTagsProvider`              |
 
-Registry objects can check their tags via `Holder#tags`, getting their `Holder` using either `Registry#getHolder` or `Registry#getHolderOrThrow`. Comparing a single tag can be done using `Holder#is`.
+Of note is the `IntrinsicHolderTagsProvider<T>` class, which is a subclass of `TagsProvider<T>` and a common superclass for `BlockTagsProvider`, `ItemTagsProvider`, `FluidTagsProvider`, `EntityTypeTagsProvider`, and `GameEventTagsProvider`. These classes (from now on called intrinsic providers for simplicity) have some additional functionality for generation that will be outlined in a moment.
 
-Tag-holding registry objects also contain a method called `#is` in either their registry object or state-aware class to check whether the object belongs to a certain tag.
+For the sake of example, let's assume that we want to generate block tags. (All other classes work the same with their respective tag types.)
 
-As an example:
 ```java
-public static final TagKey<Item> myItemTag = ItemTags.create(new ResourceLocation("mymod", "myitemgroup"));
+public class MyBlockTagsProvider extends BlockTagsProvider {
+    // Get parameters from GatherDataEvent.
+    public MyBlockTagsProvider(PackOutput output, CompletableFuture<HolderLookup.Provider> lookupProvider, ExistingFileHelper existingFileHelper) {
+        super(output, lookupProvider, ExampleMod.MOD_ID, existingFileHelper);
+    }
 
-public static final TagKey<VillagerType> myVillagerTypeTag = TagKey.create(Registries.VILLAGER_TYPE, new ResourceLocation("mymod", "myvillagertypegroup"));
+    // Add your tag entries here.
+    @Override
+    protected void addTags(HolderLookup.Provider lookupProvider) {
+        // Create a tag builder for our tag. This could also be e.g. a vanilla or NeoForge tag.
+        tag(MY_TAG)
```

Comment:
Should explain the difference between `tag` and `getOrCreateRawBuilder` since it is useful.